sys/net/netstats/netstats_neighbor.c

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright (C) Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net
+ * @file
+ * @brief       Neighbor level stats for netdev
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ * @author      Benjamin Valentin <benpicco@beuth-hochschule.de>
+ * @}
+ */
+
+#include <errno.h>
+
+#include "net/netdev.h"
+#include "net/netstats/neighbor.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static inline void _lock(netif_t *dev)
+{
+    mutex_lock(&dev->neighbors.lock);
+}
+
+static inline void _unlock(netif_t *dev)
+{
+    mutex_unlock(&dev->neighbors.lock);
+}
+
+/**
+ * @brief Compare the freshness of two records
+ *
+ * @param[in] a     pointer to the first record
+ * @param[in] b     pointer to the second record
+ * @param[in] now   current timestamp in seconds
+ *
+ * @return       pointer to the least fresh record
+ */
+static inline netstats_nb_t *netstats_nb_comp(const netstats_nb_t *a,
+                                              const netstats_nb_t *b,
+                                              uint16_t now)
+{
+    return (netstats_nb_t *)((now - a->last_updated > now - b->last_updated) ? a : b);
+}
+
+static bool l2_addr_equal(const uint8_t *a, uint8_t a_len, const uint8_t *b, uint8_t b_len)
+{
+    if (a_len != b_len) {
+        return false;
+    }
+
+    return memcmp(a, b, a_len) == 0;
+}
+
+static void netstats_nb_half_freshness(netstats_nb_t *stats, timex_t *cur)
```

Comment:
Does using `timex_t` here make sense, if you use only the lower half of the seconds part anyways in `now`?